compiler/noirc_evaluator/src/ssa/opt/array_get.rs

```diff
@@ -0,0 +1,156 @@
+use crate::ssa::{
+    ir::{
+        dfg::CallStack, function::Function, instruction::Instruction, map::Id, types::Type,
+        value::ValueId,
+    },
+    ssa_gen::Ssa,
+};
+use fxhash::FxHashMap as HashMap;
+
+impl Ssa {
+    // Given an original IfElse instruction is this:
+    //
+    //     v10 = if v0 then v2 else if v1 then v3
+    //
+    // and a later ArrayGet instruction is this:
+    //
+    //     v11 = array_get v4, index v4
+    //
+    // we optimize it to this:
+    //
+    //     v12 = array_get v2, index v4
+    //     v13 = array_get v3, index v4
+    //     v14 = if v0 then v12 else if v1 then v13
+    #[tracing::instrument(level = "trace", skip(self))]
+    pub(crate) fn array_get_optimization(mut self) -> Self {
+        for function in self.functions.values_mut() {
+            // This should match the check in flatten_cfg
+            if let crate::ssa::ir::function::RuntimeType::Brillig = function.runtime() {
+                continue;
+            }
```

Comment:
I copied this from other passes: I'm not sure what it means though!

Reply:
This just means that we only want to apply this SSA pass to functions which are being compiled to constrained ACIR rather than unconstrained Brillig. We do this as some optimisations are specific to each runtime.

This optimisation would benefit both runtimes however so we should remove this.